meson.build

```diff
@@ -0,0 +1,268 @@
+  'test/test-platform-output.c',
+  'test/test-poll.c',
+  'test/test-poll-close.c',
+  'test/test-poll-close-doesnt-corrupt-stack.c',
+  'test/test-poll-closesocket.c',
+  'test/test-poll-multiple-handles.c',
+  'test/test-poll-oob.c',
+  'test/test-process-priority.c',
+  'test/test-process-title.c',
+  'test/test-process-title-threadsafe.c',
+  'test/test-queue-foreach-delete.c',
+  'test/test-random.c',
+  'test/test-ref.c',
+  'test/test-run-nowait.c',
+  'test/test-run-once.c',
+  'test/test-semaphore.c',
+  'test/test-shutdown-close.c',
+  'test/test-shutdown-eof.c',
+  'test/test-shutdown-twice.c',
+  'test/test-signal-multiple-loops.c',
+  'test/test-signal-pending-on-close.c',
+  'test/test-signal.c',
+  'test/test-socket-buffer-size.c',
+  'test/test-spawn.c',
+  'test/test-stdio-over-pipes.c',
+  'test/test-strscpy.c',
+  'test/test-tcp-alloc-cb-fail.c',
+  'test/test-tcp-bind-error.c',
+  'test/test-tcp-bind6-error.c',
+  'test/test-tcp-close-accept.c',
+  'test/test-tcp-close-while-connecting.c',
+  'test/test-tcp-close.c',
+  'test/test-tcp-close-reset.c',
+  'test/test-tcp-create-socket-early.c',
+  'test/test-tcp-connect-error-after-write.c',
+  'test/test-tcp-connect-error.c',
+  'test/test-tcp-connect-timeout.c',
+  'test/test-tcp-connect6-error.c',
+  'test/test-tcp-flags.c',
+  'test/test-tcp-open.c',
+  'test/test-tcp-read-stop.c',
+  'test/test-tcp-read-stop-start.c',
+  'test/test-tcp-shutdown-after-write.c',
+  'test/test-tcp-unexpected-read.c',
+  'test/test-tcp-oob.c',
+  'test/test-tcp-write-to-half-open-connection.c',
+  'test/test-tcp-write-after-connect.c',
+  'test/test-tcp-writealot.c',
+  'test/test-tcp-write-fail.c',
+  'test/test-tcp-try-write.c',
+  'test/test-tcp-try-write-error.c',
+  'test/test-tcp-write-queue-order.c',
+  'test/test-test-macros.c',
+  'test/test-thread-equal.c',
+  'test/test-thread.c',
+  'test/test-threadpool-cancel.c',
+  'test/test-threadpool.c',
+  'test/test-timer-again.c',
+  'test/test-timer-from-check.c',
+  'test/test-timer.c',
+  'test/test-tmpdir.c',
+  'test/test-tty-duplicate-key.c',
+  'test/test-tty-escape-sequence-processing.c',
+  'test/test-tty.c',
+  'test/test-udp-alloc-cb-fail.c',
+  'test/test-udp-bind.c',
+  'test/test-udp-connect.c',
+  'test/test-udp-create-socket-early.c',
+  'test/test-udp-dgram-too-big.c',
+  'test/test-udp-ipv6.c',
+  'test/test-udp-mmsg.c',
+  'test/test-udp-multicast-interface.c',
+  'test/test-udp-multicast-interface6.c',
+  'test/test-udp-multicast-join.c',
+  'test/test-udp-multicast-join6.c',
+  'test/test-udp-multicast-ttl.c',
+  'test/test-udp-open.c',
+  'test/test-udp-options.c',
+  'test/test-udp-send-and-recv.c',
+  'test/test-udp-send-hang-loop.c',
+  'test/test-udp-send-immediate.c',
+  'test/test-udp-sendmmsg-error.c',
+  'test/test-udp-send-unreachable.c',
+  'test/test-udp-try-send.c',
+  'test/test-uname.c',
+  'test/test-walk-handles.c',
+  'test/test-watcher-cross-stop.c',
+  'test/runner-unix.c',
```

Comment:
This is a problem with the previous released version of the wrap too, but... how does this work, exactly? This is a stub for Unix, there is also a `test/runner-win.c`, and there's no way you can unconditionally include this... the tests run fine here only because I'm a Linux user, I think. ;)

This should include one or the other runner depending on the OS.

Reply:
The initial wrap completely ignores win32 (and other platforms) at all and is just a small working subset of the actual CMake/autoconf. I can do some improvements s.t. the meson.build comes closer to the CMakeLists.txt (which I think didn't exist when the initial wrap was created).

Reply:
I didn't check the history, but there are *currently* `if WINNT` blocks in Makefile.am, too.

Either way it would be quite good to have...

Reply:
https://mesonbuild.com/External-Project-module.html supports the `./configure; make` build system.

CMake projects are not distributable via the wrapdb, and require one to have both meson and cmake installed, so yes, ideally there is a native meson.build wrap.

Reply:
nah, i'll do it - why not. maybe they reconsider their decision in the future, then they'll have a nice starting point. follow up is coming soon :)